docs/contributing/index.rst

```diff
@@ -1,13 +1,109 @@
 Contributing
 ============
 
+.. contents::
+
 Contributing to RestrictedPython
+--------------------------------
+
+Legal requirements to contribute to RestrictedPython
+++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+The projects under the zopefoundation GitHub organization are open source, including RestrictedPython.
+We welcome contributions in different forms:
+
+* bug reports
+* code improvements and bug fixes
+* documentation improvements
+* pull request reviews
+
+For any changes in the repository besides trivial typo fixes, you are required to sign the contributor agreement.
+See https://www.zope.dev/developer/becoming-a-committer.html for details.
+
+Please visit our `Developer Guidelines`_ if you'd like to contribute code changes and our `guidelines for reporting bugs`_ if you want to file a bug report.
+
+
+.. _`Developer Guidelines`: https://www.zope.dev/developer/guidelines.html
+.. _`guidelines for reporting bugs`: https://www.zope.dev/developer/reporting-bugs.html
+
+Preperations for Contributing
++++++++++++++++++++++++++++++
+
+If you want to contribute to this package, please prepare a development environment that includes ``git``, ``tox``, and several Python versions available through a Python manager such as ``pyenv``.
+Please read the section :ref:`understand` first.
+
+For all commits, you must run the tests locally with ``tox`` before pushing to the central repository.
+
+Preperations for a new Python version
++++++++++++++++++++++++++++++++++++++
+
+RestrictedPython should be updated for each new version of Python.
+To do so:
+
+-   Read the changelog (`What's new in Python`_).
+-   Copy and adjust the new AST Grammar (found under: `Python 3 AST`_) to ``/docs/contributing/ast/python<version>.ast``.
+-   Add a new file ``changes_from<old_version>to<new_version>.rst`` in the directory ``/docs/contributing/``.
+-   Add those files to the ``toctree`` directive in ``index.rst``.
+
+For each new *AST Node*:
+
+ -   Add a ``visit_<AST Node>`` to ``/src/RestrictedPython/transformer.py``.
+ -   Add tests to ``/tests/``.
+
+If the new AST Node should be disabled by default, add a ``visit_<AST Node>`` method such as the following:
+
+.. code-block:: python
+
+    def visit_<AST Node>(self, node):
+        """`<AST Node>` expression currently not allowed."""
+        self.not_allowed(node)
```

Comment:
This is the default behavior, it is not needed to add a method just do deny using an AST node.

Reply:
I am a fan of:

> Explicit is better than implicit.

We should have a visit_<AST Node> method for each new node and also tests for every new functionallity in Python, so that we actually see if those could be a problem or a no-brainer to allow.

Reply:
I am a fan of `simple is better than complicated` and having explicit handlers makes it more complicated. `transformer.py` is already > 1200 lines of code. And we even had duplicate handlers in the past because of the sheer amount of methods in this file.

Let's keep it as simple as possible!

Reply:
I have no idea about the details here. Overall question, whats better for readability and testing, which would be most important to me?